### DISCUSSION TITLE
Dynamic link does not survive the installation from the Android Play Store.

### DISCUSSION THREAD

**Discussion by vikjovanov** [invertase/react-native-firebase#4617] [Q&A]
Hello,

when I do my tests to see if the dynamic links work well in my application, I notice that if the application is installed and in background/killed, I receive the links with the getInitialLink or the onLink. Which is great !

But when I put my application in internal test, I uninstall my application and click on a dynamic link that redirects to my application, I am correctly redirected to the store. Then, when the application is installed, the "Open" button says "Continue", but when I get to the application, I don't get the link back.

When I add '?d=1' at the end of my link and check the links that redirect to the playstore, it looks like this: 

`https://play.google.com/store/apps/details?id=appid&pcampaignid=fdl_long&url=https://mydeeplink&min_version=xx`

Here is my intent-filter for the dynamic link:

```xml
<intent-filter android:autoVerify="true">
    <action android:name="android.intent.action.VIEW"/>
    <category android:name="android.intent.category.DEFAULT"/>
    <category android:name="android.intent.category.BROWSABLE"/>
    <data android:host="example.page.link" android:scheme="http"/>
    <data android:host="example.page.link" android:scheme="https"/>
</intent-filter>
```

I read on different forums that there was no management of the REFERRER if we were not in production.
Is it a serious information ? (knowing that these are posts from quite a long time ago).
- If it is true, is there a reliable way to test the survival of a dynamic link after installation before going into production?
- If not, where can the problem come from? 

Hoping that you will be able to help me,

thank you in advance,
vikjovanov

**Comment by mikehardy** [comment]
With apologies - all the behavior here will be coming from the upstream SDK (https://github.com/firebase/firebase-android-sdk/) so I encourage you to check their issues list, and maybe try a quickstart from them directly to see if your behavior is the same - https://github.com/firebase/quickstart-android/tree/master/dynamiclinks
